src/core/bittorrent/torrenthandle.cpp

```diff
@@ -972,7 +972,7 @@ QList<PeerInfo> TorrentHandle::peers() const
     SAFE_CALL(get_peer_info, nativePeers);
 
     foreach (const libt::peer_info &peer, nativePeers)
-        peers << peer;
+        peers << PeerInfo((BitTorrent::TorrentHandle *const)this, peer);
```

Comment:
This is because of TorrentHandle::peers() is const method. In this case it is better to change PeerInfo constructor parameter type to ```const BitTorrent::TorrentHandle *``` and not to use const_cast (behind the scene).